internal/host/plugin/host_catalog_secret_test.go

```diff
@@ -118,57 +118,85 @@ func TestHostCatalogSecret_New(t *testing.T) {
 	}
 }
 
-func TestHostCatalogSecret_Custom_Queries(t *testing.T) {
-	ctx := context.Background()
+func TestHostCatalogSecret_Create_Upsert_Update_Delete(t *testing.T) {
 	conn, _ := db.TestSetup(t, "postgres")
-	rw := db.New(conn)
 	wrapper := db.TestWrapper(t)
 	kkms := kms.TestKms(t, conn, wrapper)
 
 	_, prj := iam.TestScopes(t, iam.TestRepo(t, conn, wrapper))
 	plg := host.TestPlugin(t, conn, "test")
 	cat := TestCatalog(t, conn, prj.GetPublicId(), plg.GetPublicId())
-	databaseWrapper, err := kkms.GetWrapper(ctx, prj.PublicId, kms.KeyPurposeDatabase)
-	require.NoError(t, err)
+	ctx := context.Background()
 
-	hcs, err := newHostCatalogSecret(ctx, cat.GetPublicId(),
-		&structpb.Struct{Fields: map[string]*structpb.Value{"foo": structpb.NewStringValue("bar")}})
+	secret, err := newHostCatalogSecret(ctx, cat.GetPublicId(), mustStruct(map[string]interface{}{
+		"foo": "bar",
+	}))
 	require.NoError(t, err)
-	assert.NoError(t, hcs.encrypt(ctx, databaseWrapper))
-	q, v := hcs.upsertQuery()
-	_, err = rw.Exec(ctx, q, v)
-	assert.NoError(t, err)
+	require.NotNil(t, secret)
+	require.Empty(t, secret.CtSecret)
 
-	found, err := newHostCatalogSecret(ctx, cat.GetPublicId(), nil)
+	databaseWrapper, err := kkms.GetWrapper(ctx, prj.PublicId, kms.KeyPurposeDatabase)
 	require.NoError(t, err)
-	assert.NoError(t, rw.LookupWhere(ctx, found, "catalog_id=?", found.GetCatalogId()))
+	require.NotNil(t, databaseWrapper)
+	require.NoError(t, secret.encrypt(ctx, databaseWrapper))
+
+	// Create
+	w := db.New(conn)
+	require.NoError(t, w.Create(ctx, secret))
+
+	// Upsert
+	newStructUpsert := mustMarshal(map[string]interface{}{
+		"baz": "qux",
+	})
+	newSecretUpsert := secret.clone()
+	newSecretUpsert.Secret = newStructUpsert
+	require.NoError(t, newSecretUpsert.encrypt(ctx, databaseWrapper))
+	require.NoError(t, w.Create(ctx, newSecretUpsert, db.WithOnConflict(&db.OnConflict{
+		Target: db.Columns{"catalog_id"},
+		Action: db.SetColumnValues(map[string]interface{}{
+			"secret": newSecretUpsert.CtSecret,
+			"key_id": newSecretUpsert.KeyId,
+		}),
```

Comment:
Instead of using SetColumnValues, which require you passing in the values into the map, consider this potentially more succinct Action.  Here and in `repository_host_catalog.go`
```suggestion
		Action: db.SetColumns([]string{"secret", "key_id"}),
```

Reply:
Done!